### PR TITLE
Add Info windows for Ensemble, Node and Network

### PR DESCRIPTION
**Pull request by tbekolay**
We think that this is all of the NetGraphItem types (if not, issue for future PR).

These are all accessible through a menu item that always shows at the bottom as "Details ...". Clicking on the details buttons brings up a modal window with info about that object. If a simulation is currently running, we pause it before bringing up the modal, and replay when the modal closes.

For Ensembles, we show parameter info, tuning / response curves (if and only if the model's been built) and connection info.

screen shot 2015-05-21 at 12 17 58 pm

screen shot 2015-05-21 at 12 18 10 pm

screen shot 2015-05-21 at 12 18 21 pm

For Nodes, we show parameter info, the node's output for the first 1.0 seconds (if `size_in == 0`, and the output is not None), and connection info.

screen shot 2015-05-21 at 12 19 12 pm

screen shot 2015-05-21 at 12 19 18 pm

screen shot 2015-05-21 at 12 19 26 pm

For Networks, we show some summary statistics, and connection info.

screen shot 2015-05-21 at 12 20 15 pm

screen shot 2015-05-21 at 12 20 22 pm

### CODE REVIEW

File: examples/net.py [hunk 1]
```diff
@@ -12,19 +12,21 @@
     nengo.Connection(ens, result, function=lambda x: x[0] * x[1],
                      synapse=0.01)
 
-    with nengo.Network(label='subnet') as subnet:
+    with nengo.Network(label='mysubnet') as subnet:
```
Comment (tcstewar) [inline examples/net.py]: Can we remove these changes from the PR?  It's not necessary, but no sense changing other people's examples

Comment (tbekolay) [inline examples/net.py]: Whoops! I think these were accidentally included. I'll amend accordingly.

File: nengo_viz/static/viz_sim_control.js [hunk 2]
```diff
@@ -9,17 +9,17 @@
...
 
-    div.classList.add('sim_control');    
+    div.classList.add('sim_control');
     this.div = div;
     div.style.position = 'fixed';
-    //TODO: Is this the method we want to use for keeping simcontrol on top 
+    //TODO: Is this the method we want to use for keeping simcontrol on top
     //or do we want to dynamically raise it's zindex when other elements have theres increased
-    div.style.zIndex = 99999999; 
+    div.style.zIndex = 1040;
```
Comment (tcstewar) [inline nengo_viz/static/viz_sim_control.js]: Why was this changed?

File: examples/net.py (hunk 1 above)
Comment (tbekolay) [inline examples/net.py]: I'll get rid of the `viz.js` changes also, that too was an accident ;)

File: nengo_viz/static/viz_sim_control.js (hunk 2 above)
Comment (tbekolay) [inline nengo_viz/static/viz_sim_control.js]: Bootstrap's modal has a hard-coded z-index of 1050. While we can change that for the div itself, it wasn't obvious to me how to change it for the shaded background, so unless we change this z-index, the sim_control will show up under the modal, but over the background shading. It could be possible to dig a bit deeper and figure out how to set the background z-index higher; do we expect to get z-indexes over 1050 anywhere?

Comment (tcstewar) [inline nengo_viz/static/viz_sim_control.js]: Every time you click on something, the max_zindex increases.  So if you click on a 1000 things, items will start showing up on top of the modal.  :(

https://github.com/nengo/nengo_viz/blob/master/nengo_viz/static/viz.js#L82

Comment (tbekolay) [inline nengo_viz/static/viz_sim_control.js]: Mkay, I'll look into setting the modal backdrop z-index!

**Comment by tbekolay** [comment]
OK, I think I addressed all the feedback. The z-index of the sim control is no longer changed, and the whitespace changes are gone (I think!)

File: nengo_viz/static/viz_tooltips.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+    $tooltip.popover({"trigger": "hover"});
+}
+
+/* Ensemble Parameters */
+VIZ.tooltips.ens = {};
+VIZ.tooltips.ens.n_neurons = ["Type: int", "The number of neurons."];
+VIZ.tooltips.ens.dimensions = [
+    "Type: int", "The number of representational dimensions."];
+VIZ.tooltips.ens.radius = [
+    "Type: int\nDefault: 1.0", "The representational radius of the ensemble."];
```
Comment (tcstewar) [inline nengo_viz/static/viz_tooltips.js]: float  :)

File: nengo_viz/static/viz_modal.js [hunk 4]
```diff
@@ -0,0 +1,415 @@
...
+    var tabs = info_body([{id: 'stats', title: 'Statistics'},
+                          {id: 'connections', title: 'Connections'}]);
+    render_stats(tabs.stats, stats);
+    render_connections(tabs.connections, uid, conninfo);
+}
+
+/**
+ * Sets up the tabs for Info modals.
+ */
+function info_body(tabinfo) {
```
Comment (tcstewar) [inline nengo_viz/static/viz_modal.js]: These functions should be somewhere in the VIZ namespace (perhaps ```VIZ.Modal.info_body```)

Comment (tbekolay) [inline nengo_viz/static/viz_modal.js]: Will do!

**Comment by tbekolay** [comment]
OK, I've rebased this to master, and put all of the functions in the `VIZ` namespace. Should be ready for review.

**Comment by tcstewar** [comment]
Looks good to me!
